### PR TITLE
Fixed the inability to serve static files greater than 140 KB

### PR DESCRIPTION
**Pull request by jeffvandyke**
Added offset_ to BufferHolder so that retries would remember where they left off,
Fixed range of copying in struct Buffer so that large dynamic responses still work.

### CODE REVIEW

**Comment by jeffvandyke** [comment]
This fixes my specific issue mentioned on #35, let me know if there's any thing small that needs to be done before this gets merged in, because this seems like a pretty serious bug.

**Comment by oktal** [comment]
Hello,

First of all, thanks for taking time to fix this issue. If I remember correctly, someone also already raised the issue and came to the same conclusion (basically, we do not keep the offset of the last byte written in the buffer).

Your fix looks perfectly fine to me. I'll take some additional time to review it properly this week-end, and merge it in.

**Comment by jeffvandyke** [comment]
Thanks! It was a team effort over here, to give proper credit.

**Comment by jeffvandyke** [comment]
Say, just wondering if you had time to merge this pull request, and perhaps some others, into master.

**Comment by jeffvandyke** [comment]
Say, hate to nudge, but it's been a little bit, is this fix going to get merged? It seems like a potentially big issue to me. Or is there something I can do to help?

**Comment by oktal** [comment]
Hello,

Sorry for that, I did not find time yet, but will try to find some this week-end.

**Comment by jeffvandyke** [comment]
That's alright, thanks! and let me know if you find anything worth improving.

File: src/common/transport.cc [hunk 1]
```diff
@@ -200,6 +200,7 @@ Transport::asyncWriteImpl(
         if (bytesWritten < 0) {
             if (errno == EAGAIN || errno == EWOULDBLOCK) {
                 if (status == FirstTry) {
+                    toWrite.erase(fd);
```
Comment (oktal) [inline src/common/transport.cc]: Could you please explain why this `erase` is needed ? It's not clear to me.

Basically, if it's the first time we're hitting `EAGAIN`, the fd should not be in the map. Maybe is it not cleaned afterwise ?

**Comment by jeffvandyke** [comment]
I'm going to have to say good catch :) I was a bit fuzzy on when `toWrite` had the `fd` in it, so I had written it to set status to FirstTry on any successful write, so it would always erase, even if not there, and then add the entry with the offset saved to the buffer. I'm now always inserting the entry on a `EAGAIN` and only erasing it if status == Retry.

I've made a new small commit that has this fix implemented that's ready to review.

**Comment by lucas-bremond** [comment]
Thank you @jeffvandyke, I've merged your changes into my local fork and it is now working like a charm.

**Comment by oktal** [comment]
Great, thanks @lucas-bremond for the feedback.

I've made some changes recently that are now in conflict with your changes. Do you want to rebase your changes on top of the new master @jeffvandyke ?

Hopefully I'll have some free time this week-end to close this PR. Kudos again

**Comment by jeffvandyke** [comment]
Sure, I'll try to get around to it before this weekend. Thanks!

**Comment by jeffvandyke** [comment]
Alright, sorry it's been a while, but I've finally got my changes rebased. I tested it with a 300 KB file and it still works.

**Comment by oktal** [comment]
Alright, great job, merging-it in !

**Pull request merged**